contracts/modules/Experimental/STO/POLYPeggedSTO.sol

```diff
@@ -0,0 +1,572 @@
+    }
+
+    /**
+     * @dev modifies max non accredited investment limit and overall minimum investment limit
+     * @param _minimumInvestment overall minimum investment limit in USD
+     * @param _nonAccreditedLimit max non accredited investment limit in USD
+     * @param _maxNonAccreditedInvestors Maximum number of non-accredited investors allowed (0 = unlimited)
+     */
+    function modifyLimits(
+        uint256 _minimumInvestment,
+        uint256 _nonAccreditedLimit,
+        uint256 _maxNonAccreditedInvestors
+    ) external notStarted {
+        _onlySecurityTokenOwner();
+        _modifyLimits(_minimumInvestment, _nonAccreditedLimit, _maxNonAccreditedInvestors);
+    }
+
+    /**
+     * @dev Modifies USD rate per token
+     * @param _rate How many token units a buyer gets per USD multiplied by 10^18
+     */
+    function modifyRate(uint256 _rate) external notStarted {
+        _onlySecurityTokenOwner();
+        _modifyRate(_rate);
+    }
+
+    /**
+     * @dev Modifies how many token base units this STO will be allowed to sell to investors
+     * @param _cap Max number of token that can be sold by 10^18 per base unit
+     */
+    function modifyCap(uint256 _cap) external notStarted {
+        _onlySecurityTokenOwner();
+        _modifyCap(_cap);
+    }
+
+    /**
+     * @dev Modifies STO start and end times
+     * @param _startTime start time of sto
+     * @param _endTime end time of sto
+     */
+    function modifyTimes(uint256 _startTime, uint256 _endTime) external notStarted {
+        _onlySecurityTokenOwner();
+        _modifyTimes(_startTime, _endTime);
+    }
+
+    /**
+     * @dev Modifies addresses used as wallet and treasury wallet
+     * @param _wallet Address of wallet where funds are sent
+     */
+    function modifyWalletAddress(address payable _wallet) external {
+        //can be modified even when STO started
+        _onlySecurityTokenOwner();
+        _modifyWalletAddress(_wallet);
+    }
+
+    /**
+     * @notice Use to change the treasury wallet
+     * @param _treasuryWallet Ethereum account address to receive unsold tokens
+     * @notice Set to address zero to use dataStore treasuryWallet
+     */
+    function modifyTreasuryWallet(address _treasuryWallet) external {
+        //can be modified even when STO started
+        _onlySecurityTokenOwner();
+        _modifyTreasuryWallet(_treasuryWallet);
+    }
+
+    function _modifyLimits(
+        uint256 _minimumInvestment,
+        uint256 _nonAccreditedLimit,
+        uint256 _maxNonAccreditedInvestors
+    ) internal {
+        minimumInvestment = _minimumInvestment;
+        nonAccreditedLimit = _nonAccreditedLimit;
+        maxNonAccreditedInvestors = _maxNonAccreditedInvestors;
+        emit SetLimits(minimumInvestment, nonAccreditedLimit, maxNonAccreditedInvestors);
+    }
+
+    function _modifyRate(uint256 _rate) internal {
+        require(_rate > 0, "Rate not > 0");
```

Comment:
Try to write the reason strings in a way that the developer understands what went wrong. Not what will be the valid value for that variable. (If there is a size issue then it can be considered).

Reply:
I've changed a few. Let me know if there is a specific one that you'd like changed. I've them here for your convenience:
- STO already started
- Already configured
- Invalid rate
- Invalid cap
- Invalid times
- Invalid wallet
- Already finalized
- Invalid treasury address
- Length mismatch
- Does not change value
- Beneficiary is not funder
- Unauthorized beneficiary
- Transfer failed
- STO not open
- No funds sent
- Invalid beneficiary
- Less than minimum investment
- Investment limit reached
- Max non-accredited investors
- Insufficient tokens minted
- Invalid oracle
- No tokens

Reply:
You could change this `Does not change value` to be more specific.

Reply:
It's hard to make this more specific and and try to keep the length short with the variable name being 27 characters long. Would you prefer `Does not change allowBeneficialInvestments` or `allowBeneficialInvestments unchanged` which both exceed the length of a 32 byte string? 

Given that the function `changeAllowBeneficialInvestments` would typically be called as a stand alone function with only a single input variable it should be clear to a developer why it reverted so may not need to be more specific.

For reference USDTieredSTO.sol uses `Value unchanged` while CappedSTO.sol also uses `Does not change value` for the same function.